### PR TITLE
Issue 3923 memory leak fixes

### LINKED ISSUE #3923: Locate memory leak in jest
### Description
Jest is occasionally erroring out due to running out of memory, we should investigate on where this is happening and plug it

### Suggested solution
Locate memory leak and fix it

Some guidance here that may be useful: https://chanind.github.io/javascript/2019/10/12/jest-tests-memory-leak.html 

### Goal
- jest should no longer error out due to out of heap space

### Tasks
- [x] Run unit tests without `forceExit`
- [x] See if we can run driver/e2e/script tests without `forceExit`
- [x] fix v4 tests
- [ ] Inspect memory

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #3923

#### Description
- ensure all services are cleaned up after test so `--forceExit` is no longer necessary on all tests
  - make v4 queue and db handler use v5's connection
  - make an async version of creating express app
- some fixes on tests due to the change above
- wipe data on database after every e2e test group

#### Test cases
tests should still pass as before
Memory seems to stablise (with debugging on GC exposed)
image

### CODE REVIEW

**Review by ChristosTsiotsias — CHANGES_REQUESTED** [review]

File: backend/tests/v5/unit/models/loginRecords.test.js [hunk 1]
```diff
@@ -243,13 +235,11 @@ const testIsAccountLocked = () => {
...
 	describe('Remove all user login records', () => {
 		test('Should just drop the user collection within loginRecords', async () => {
-			const fn = jest.spyOn(db, 'deleteMany').mockResolvedValue(undefined);
-
 			const user = generateRandomString();
 			await expect(LoginRecord.removeAllUserRecords(user)).resolves.toBeUndefined();
 
-			expect(fn).toHaveBeenCalledTimes(1);
-			expect(fn).toHaveBeenCalledWith(INTERNAL_DB, loginRecordsCol, { user });
+			expect(DBHandler.deleteMany).toHaveBeenCalledTimes(1);
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply backend/tests/v5/unit/models/loginRecords.test.js:241]: same as above

File: backend/tests/v5/unit/models/loginRecords.test.js [hunk 2]
```diff
@@ -258,24 +248,22 @@ const testGetLastLoginDate = () => {
...
 
-			expect(fn).toHaveBeenCalledTimes(1);
-			expect(fn).toHaveBeenCalledWith(INTERNAL_DB, loginRecordsCol,
+			expect(DBHandler.findOne).toHaveBeenCalledTimes(1);
+			expect(DBHandler.findOne).toHaveBeenCalledWith(INTERNAL_DB, loginRecordsCol,
 				{ user, failed: { $ne: true } }, { loginTime: 1 }, { loginTime: -1 });
 		});
 
 		test('should return undefined if the user has no login record', async () => {
-			const fn = jest.spyOn(db, 'findOne').mockResolvedValueOnce(undefined);
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply backend/tests/v5/unit/models/loginRecords.test.js:272]: same

File: backend/tests/v5/unit/models/modelSettings.test.js [hunk 3]
```diff
@@ -189,27 +183,22 @@ const testGetFederations = () => {
...
+			expect(DBHandler.find).toHaveBeenCalledWith(teamspace, SETTINGS_COL,
+				{ _id: { $in: modelIds }, federate: true }, undefined, undefined);
 		});
 	});
 };
 
 const testAddModel = () => {
 	describe('Add model', () => {
 		test('should return inserted ID on success when a container is added', async () => {
-			const fn = jest.spyOn(db, 'insertOne');
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply backend/tests/v5/unit/models/modelSettings.test.js:211]: ditto 😆

File: backend/tests/v5/unit/utils/quota.test.js [hunk 4]
```diff
@@ -85,7 +88,7 @@ const testGetQuotaInfo = () => {
 		[tsWithFreeQuota]: {},
 	};
 
-	jest.spyOn(db, 'findOne').mockImplementation((ts, col, { _id }) => {
+	DBHandler.findOne.mockImplementation((ts, col, { _id }) => {
```
[the remark this answers is not in the document]
Comment (carmenfan) [reply backend/tests/v5/unit/utils/quota.test.js:91]: This is serving every single tests within the group 
image

**Review by ChristosTsiotsias — APPROVED** [review]

**Pull request merged**